charts/gardener/templates/rbac.yaml

```diff
@@ -137,6 +137,43 @@ subjects:
   kind: Group
   name: system:authenticated
 
+# Cluster role with cluster role binding allowing all authenticated users to create projects
+---
+apiVersion: {{ include "rbacversion" . }}
+kind: ClusterRole
+metadata:
+  name: garden.sapcloud.io:system:projects
+  labels:
+    app: gardener
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+rules:
+- apiGroups:
+  - garden.sapcloud.io
+  resources:
+  - projects
+  verbs:
+  - create
+---
+apiVersion: {{ include "rbacversion" . }}
+kind: ClusterRoleBinding
+metadata:
+  name: garden.sapcloud.io:system:projects
+  labels:
+    app: gardener
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: garden.sapcloud.io:system:projects
+subjects:
+- apiGroup: rbac.authorization.k8s.io
+  kind: Group
+  name: system:authenticated
```

Comment:
This is **extremely**  broad as it'll allow for every service account token to be used to create `Projects`, then `Shoots` and greatly escalating the privileges of every single user or software component. 

This ClusterRoleBinding should
- be created by the Dashboard for every single `ServiceAccount

or

- be limited to specific `Users` and/or `Groups`

Reply:
I'm not really understanding the point here. Are you criticizing that technical users aka service accounts can create `Project`s now? What do you mean by "escalating the privileges of every single user"?

> be created by the Dashboard for every single `ServiceAccount`

We cannot assume that the Dashboard is deployed or actually can be deployed at all.

> be limited to specific `Users` and/or `Groups`:

We also can't do that because we don't know anything about the connected OIDC provider, especially not which `Users` or `Groups` do exist and to which of them we should allow creating projects.

Can you explain again what your doubts are here?

Reply:
> Are you criticizing that technical users aka service accounts can create Projects now?

If you are bound ONLY to this `ClusterRole`

```yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: system:discovery
rules:
- nonResourceURLs:
  - /api
  - /api/*
  - /apis
  - /apis/*
  - /healthz
  - /openapi
  - /openapi/*
  - /swagger-2.0.0.pb-v1
  - /swagger.json
  - /swaggerapi
  - /swaggerapi/*
  - /version
  - /version/
  verbs:
  - get
```

then with this new rolebinding you'll also have access to create `Projects` which might not be what cluster administrators want to have. 

`ServiceAccount` with name `system:serviceaccount:foo:bob-the-robot` or `Group` `administrators` - yes, definitely. `User` `system:nodes:node-foo` - most likely no. Any authenticated user - definitely no.

I'm opposed giving every single authenticated request the possibility to create those resources.

Reply:
OK, I've synched with @swapnilgm and @vlerenc and I will do the following changes: I will remove the `ClusterRoleBinding` currently allowing all authenticated users to create projects completely. We will keep it in the responsibility of the person deploying Gardener to define that `ClusterRoleBinding` properly.
In our case we are planning to keep `Project` creation restricted to the Dashboard's service account, for example. OK?

Reply:
Sounds great.